File: core/service/token_service.go
package service

import (
	"encoding/base64"
	"encoding/json"
	"time"

	"github.com/nexus-flow/nexus/core/model"
	"github.com/nexus-flow/nexus/core/repository"
)

type TokenDTO struct {
	ID        int64      `json:"id"`
	TokenDesc string     `json:"tokenDesc"`
	CreatedAt *time.Time `json:"createdAt,omitempty"`
}

func CreateToken(tokenDesc string, userID int64) (string, error) {
	vo := map[string]interface{}{
		"userId":    userID,
		"timestamp": time.Now().UnixMilli(),
	}
	data, _ := json.Marshal(vo)
	tokenValue := base64.URLEncoding.EncodeToString(data)

	t := &model.Token{
		TokenValue: tokenValue,
		TokenDesc:  tokenDesc,
	}
	t.CreatedBy = &userID
	if err := repository.CreateToken(t); err != nil {
		return "", err
	}
	return tokenValue, nil
}

func DeleteToken(id int64) error {
	return repository.DeleteToken(id)
}

func PageTokens(p *model.PageParam) ([]TokenDTO, int64, error) {
	list, total, err := repository.PageTokens(p)
	if err != nil {
		return nil, 0, err
	}
	dtos := make([]TokenDTO, 0, len(list))
	for _, t := range list {
		dtos = append(dtos, TokenDTO{
			ID:        t.ID,
			TokenDesc: t.TokenDesc,
			CreatedAt: t.CreatedAt,
		})
	}
	return dtos, total, nil
}

File: server/src/main/java/com/javaoffers/nexus/core/service/TokenService.java
package com.javaoffers.nexus.core.service;

import com.javaoffers.nexus.core.model.PageParam;
import com.javaoffers.nexus.core.model.Token;
import com.javaoffers.nexus.core.repository.TokenMapper;
import com.javaoffers.nexus.pkg.exception.BizException;
import lombok.Data;
import org.springframework.stereotype.Service;

import java.nio.charset.StandardCharsets;
import java.util.ArrayList;
import java.util.Base64;
import java.util.Date;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/** 对应 Go core/service/token_service.go。 */
@Service
public class TokenService {

    private final TokenMapper tokenMapper;

    public TokenService(TokenMapper tokenMapper) {
        this.tokenMapper = tokenMapper;
    }

    @Data
    public static class TokenDTO {
        private Long id;
        private String tokenDesc;
        private Date createdAt;
    }

    public String createToken(String tokenDesc, Long userID) {
        Map<String, Object> vo = new HashMap<>();
        vo.put("userId", userID);
        vo.put("timestamp", System.currentTimeMillis());
        String tokenValue;
        try {
            String json = new com.fasterxml.jackson.databind.ObjectMapper().writeValueAsString(vo);
            tokenValue = Base64.getUrlEncoder().encodeToString(json.getBytes(StandardCharsets.UTF_8));
        } catch (Exception e) {
            throw new BizException("生成Token失败");
        }
        Token t = new Token();
        t.setTokenValue(tokenValue);
        t.setTokenDesc(tokenDesc);
        t.setCreatedBy(userID);
        tokenMapper.create(t);
        return tokenValue;
    }

    public void deleteToken(Long id) {
        tokenMapper.delete(id);
    }

    public Page<TokenDTO> pageTokens(PageParam p) {
        int pn = p.getPageNum() <= 0 ? 1 : p.getPageNum();
        int ps = p.getPageSize() <= 0 ? 10 : p.getPageSize();
        List<Token> list = tokenMapper.page(pn, ps);
        long total = tokenMapper.count();
        List<TokenDTO> dtos = new ArrayList<>();
        for (Token t : list) {
            TokenDTO dto = new TokenDTO();
            dto.setId(t.getId());
            dto.setTokenDesc(t.getTokenDesc());
            dto.setCreatedAt(t.getCreatedAt());
            dtos.add(dto);
        }
        return new Page<>(total, dtos);
    }
}
